File: services/backend/src/models/Video.js
const mongoose = require('mongoose');

const videoSchema = new mongoose.Schema(
  {
    author: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    caption: { type: String, default: '', maxlength: 500 },
    videoUrl: { type: String, required: true },
    thumbnailUrl: { type: String, default: '' },
    duration: { type: Number, default: 0 }, // بالثواني
    width: Number,
    height: Number,
    soundName: { type: String, default: '' },
    hashtags: [{ type: String, lowercase: true, trim: true }],
    mentions: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
    likesCount: { type: Number, default: 0 },
    commentsCount: { type: Number, default: 0 },
    sharesCount: { type: Number, default: 0 },
    viewsCount: { type: Number, default: 0 },
    isHidden: { type: Boolean, default: false },
    isDeleted: { type: Boolean, default: false },
    visibility: { type: String, enum: ['public', 'followers', 'private'], default: 'public' },
  },
  { timestamps: true },
);

videoSchema.index({ hashtags: 1 });
videoSchema.index({ createdAt: -1 });

module.exports = mongoose.model('Video', videoSchema);

File: apps/mobile/lib/features/short_video/domain/entities/video.dart
import 'package:equatable/equatable.dart';
import '../../../auth/domain/entities/user.dart';

class Video extends Equatable {
  final String id;
  final User author;
  final String caption;
  final String videoUrl;
  final String thumbnailUrl;
  final int likesCount;
  final int commentsCount;
  final int sharesCount;
  final int viewsCount;
  final List<String> hashtags;

  const Video({
    required this.id,
    required this.author,
    required this.videoUrl,
    this.caption = '',
    this.thumbnailUrl = '',
    this.likesCount = 0,
    this.commentsCount = 0,
    this.sharesCount = 0,
    this.viewsCount = 0,
    this.hashtags = const [],
  });

  @override
  List<Object?> get props => [id, videoUrl, likesCount];
}